addons/pos_sale/models/pos_order.py

```diff
@@ -32,8 +32,11 @@ def _prepare_invoice_vals(self):
         invoice_vals = super(PosOrder, self)._prepare_invoice_vals()
         invoice_vals['team_id'] = self.crm_team_id.id
         addr = self.partner_id.address_get(['delivery'])
-        invoice_vals['partner_shipping_id'] = addr['delivery']
         sale_orders = self.lines.mapped('sale_order_origin_id')
+        if sale_orders and (sale_orders.partner_invoice_id.id != sale_orders.partner_shipping_id.id):
+            invoice_vals['partner_shipping_id'] = sale_orders.partner_shipping_id.id
+        else:
+            invoice_vals['partner_shipping_id'] = addr['delivery']
```

Comment:
```suggestion
            addr = self.partner_id.address_get(['delivery'])
            invoice_vals['partner_shipping_id'] = addr['delivery']
```
Remove the `address_get` call above and move it inside the else clause in your diff.